e2e/default_test.go

```diff
@@ -89,6 +96,18 @@ func TestDefaultSetup(t *testing.T) {
 					{ObjectMeta: metav1.ObjectMeta{Name: testSpinAppName, Namespace: testNamespace}},
 				},
 			}
+			res, err := resources.New(client.RESTConfig())
+			if err != nil {
+				t.Fatalf("Could not create controller runtime client: %s", err)
+			}
+			var deploy appsv1.Deployment
+			if err = res.Get(ctx, testSpinAppName, testNamespace, &deploy); err != nil {
+				t.Fatalf("Could not find deployment: %s", err)
+			}
+			v, ok := deploy.Spec.Template.Labels[testPodLabelName]
```

Comment:
Should this potentially be a new test - with default_test remaining relatively minimal, so we can keep the basics separate from customizations?

Reply:
@endocrimes I've pushed another commit to this PR.

I moved creation of the `RuntimeClass` to the test suite setup to prevent myself from adding setup and teardown steps for every test scenario.

Right now the `PodLabel` test is re-using the `SpinAppExecutor` which has been deployed as part of the default_test. 

How would you treat this one? Should 

1. every scenario setup and teardown its own `SpinAppExectuor` (maybe even its own `Namespace`)
2. there be a dedicated test for the `SpinAppExecutor` which could be re-used by all following tests

The second one would require to ensure `SpinAppExecutor` test runs before all other tests in the module (is that possible? Haven't used Kubernetes e2e-framework till today)

Reply:
In similar testing scenarios I've followed something similar to https://github.com/kubernetes-sigs/e2e-framework/tree/main/examples/every_test_custom_ns and done those things in the before/after{each test} blocks